python/aibrix_kvcache/aibrix_kvcache/cache_handle.py

```diff
@@ -67,3 +69,112 @@ def release(self) -> None:
 
     def __len__(self) -> int:
         return len(self._mrs)
+
+
+class GDRKVCacheHandle(KVCacheHandle):
+    def __init__(
+        self,
+        mr_dtype: torch.dtype,
+        mr_shape: Tuple[int, ...],
+        mrs: Sequence[Sequence[MemoryRegion]],
+    ) -> None:
+        self._mr_dtype = mr_dtype
+        self._mr_shape = mr_shape
+        self._mrs = mrs
+
+    @property
+    def memory_regions(self) -> Sequence[Sequence[MemoryRegion]]:
+        return self._mrs
+
+    def to_tensors(self) -> Sequence[Sequence[torch.Tensor]]:
+        return [
+            MemoryRegion.to_tensors(mrs, self._mr_dtype, self._mr_shape)
+            for mrs in self._mrs
+        ]
+
+    def release(self) -> None:
+        for block_mrs in self._mrs:
+            for mr in block_mrs:
+                mr.ref_down()
+
+    def __len__(self) -> int:
+        return len(self._mrs)
+
+    @staticmethod
+    def create(
+        blocks: Sequence[torch.Tensor],
+        block_spec: KVCacheBlockSpec,
+        slot_mapping: Sequence[int] | torch.Tensor,
+    ) -> KVCacheHandle:
+        """
+        Create a KVCacheHandle for the given token list.
+
+        Args:
+            blocks: Blocks to be scattered/gathered. Dim: (num_layers, 2,
+                    num_blocks, block_size, num_kv_heads, head_size)
+            block_spec: KVCache block spec.
+            slot_mapping: Mapping from tokens to blocks.
+
+        Returns:
+            KVCacheHandle.
+        """
+        assert block_spec.engine_block_ntokens == block_spec.block_ntokens, (
+            "engine KVCache and AIBrix KVCache should use the same "
+            "block_size to enable GDR"
+        )
+        block_ntokens = block_spec.block_ntokens
+        block_nbytes = block_spec.block_nbytes
+        block_dtype = block_spec.block_dtype
+        block_shape = block_spec.block_shape
+        kvcache_block_layout = block_spec.block_layout
+
+        num_layers = len(blocks)
+        assert num_layers > 0, "blocks must not be empty"
+
+        ntokens = (
+            len(slot_mapping)
```

Comment:
i am curious why not to convert slot_mapping to tensor directly so we do not need to check `if/else` here.  keep it as sequence help in some cases?